### PR TITLE
Doc update core:map class

### PR DESCRIPTION
**Pull request by jpsiitonen**
Updating example codes for map methods usage.
Currently there are no updates for Core:Map landing page ( already documented properly )

### CODE REVIEW

File: sdk/lib/core/map.dart [hunk 1]
```diff
@@ -51,13 +51,25 @@ abstract class Map<K, V> {
   /// A `LinkedHashMap` requires the keys to implement compatible
   /// `operator==` and `hashCode`.
   /// It iterates in key insertion order.
+  /// ```dart
+  /// final data = {1: 'A', 2: 'B', 3: 'C'};
```
Comment (tomgilder) [inline sdk/lib/core/map.dart]: I think we should use something other than 123/ABC - let's discuss

File: sdk/lib/core/map.dart [hunk 2]
```diff
@@ -51,13 +51,25 @@ abstract class Map<K, V> {
...
   /// Creates a [LinkedHashMap] with the same keys and values as [other].
   ///
   /// A `LinkedHashMap` requires the keys to implement compatible
   /// `operator==` and `hashCode`, and it allows `null` as a key.
   /// It iterates in key insertion order.
+  /// ```dart
+  /// final data = {1: 'A', 2: 'B', 3: 'C'};
+  /// final mapOf = Map.of(data);
+  /// print(mapOf); // {1: A, 2: B, 3: C}
+  /// print(mapOf.runtimeType); // LinkedHashMap
```
Comment (tomgilder) [inline sdk/lib/core/map.dart]: With all of these examples the `runtimeType` has generic params 

So this prints `LinkedHashMap<int, String>` - I think we should be accurate with that?

File: sdk/lib/core/map.dart [hunk 3]
```diff
@@ -134,10 +153,11 @@ abstract class Map<K, V> {
   /// The map construction iterates over [keys] and [values] simultaneously,
   /// and adds an entry to the map for each pair of key and value.
   /// ```dart
-  /// List<String> letters = ['b', 'c'];
-  /// List<String> words = ['bad', 'cat'];
-  /// var map = Map.fromIterables(letters, words);
-  /// // map is {"b": "bad", "c": "cat"}
+  /// final letters = ['b', 'c'];
+  /// final words = ['bad', 'cat'];
```
Comment (tomgilder) [inline sdk/lib/core/map.dart]: why is the cat bad 😢

File: sdk/lib/core/map.dart [hunk 4]
```diff
@@ -211,12 +238,22 @@ abstract class Map<K, V> {
   ///
   /// Returns true if any of the values in the map are equal to `value`
   /// according to the `==` operator.
+  /// ```dart
+  /// final data = {'A': 'X', 'B': 'Y', 'C': 'Z'};
+  /// final isA = data.containsValue('A'); // false
```
Comment (tomgilder) [inline sdk/lib/core/map.dart]:
```suggestion
  /// final containsA = data.containsValue('A'); // false
```

File: sdk/lib/core/map.dart [hunk 5]
```diff
@@ -211,12 +238,22 @@ abstract class Map<K, V> {
...
+  /// final isZ = data.containsValue('Z'); // true
```
Comment (tomgilder) [inline sdk/lib/core/map.dart]:
```suggestion
  /// final containsZ = data.containsValue('Z'); // true
```

File: sdk/lib/core/map.dart [hunk 6]
```diff
@@ -211,12 +238,22 @@ abstract class Map<K, V> {
...
+  /// ```
   bool containsValue(Object? value);
 
   /// Whether this map contains the given [key].
   ///
   /// Returns true if any of the keys in the map are equal to `key`
   /// according to the equality used by the map.
+  /// ```dart
+  /// final data = {'A': 'X', 'B': 'Y', 'C': 'Z'};
+  /// final isA = data.containsKey('A'); // true
```
Comment (tomgilder) [inline sdk/lib/core/map.dart]:
```suggestion
  /// final containsA = data.containsKey('A'); // true
```

File: sdk/lib/core/map.dart [hunk 7]
```diff
@@ -211,12 +238,22 @@ abstract class Map<K, V> {
...
+  /// final isZ = data.containsKey('Z'); // false
```
Comment (tomgilder) [inline sdk/lib/core/map.dart]:
```suggestion
  /// final containsZ = data.containsKey('Z'); // false
```

File: sdk/lib/core/map.dart [hunk 8]
```diff
@@ -248,6 +285,26 @@ abstract class Map<K, V> {
   ///
   /// The operation is equivalent to doing `this[entry.key] = entry.value`
   /// for each [MapEntry] of the iterable.
+  /// ```dart
+  /// final map1 = {'A': 'X', 'B': 'Y', 'C': 'Z'};
+  /// map1.addEntries([const MapEntry('E', 'AB'), const MapEntry('F', 'CD')]);
+  /// print(map1); // {A: X, B: Y, C: Z, E: AB, F: CD}
```
Comment (tomgilder) [inline sdk/lib/core/map.dart]: I just can't follow these letters easily, needs to be something else easier to follow

File: sdk/lib/core/map.dart [hunk 9]
```diff
@@ -248,6 +285,26 @@ abstract class Map<K, V> {
...
+  ///
+  /// final map2 = {1: 'A', 2: 'B', 3: 'C'};
+  ///
+  /// final customMap = {};
+  /// customMap['map1'] = map1;
+  /// customMap['map2'] = map2;
+  /// customMap['value'] = 123;
+  /// customMap['text'] = 'foo';
+  /// print(customMap);
```
Comment (tomgilder) [inline sdk/lib/core/map.dart]: I don't think we need these examples of using the standard Map syntax

File: sdk/lib/core/map.dart [hunk 10]
```diff
@@ -261,15 +318,35 @@ abstract class Map<K, V> {
...
   /// If the key is not present, [ifAbsent] must be provided.
+  /// ```dart
+  /// final data = {1: 'A', 2: 'B', 3: 'C'};
+  /// data.update(2, (value) => 'ABCD', ifAbsent: () => 'G');
+  /// print(data); // {1: A, 2: ABCD, 3: C}
+  /// // if there is no key then ifAbsent is used
+  /// data.update(10, (value) => 'XYZ', ifAbsent: () => 'X');
+  /// print(data); // {1: A, 2: ABCD, 3: C, 10: X}
+  /// ```
   V update(K key, V update(V value), {V ifAbsent()?});
```
Comment (tomgilder) [inline sdk/lib/core/map.dart:320]: This example needs work, and I think the text does too - me and Alex both weren't clear on what this method does.

- The text doesn't explain what `ifAbsent` actually does
- What happens if they key doesn't exist and you don't provide `ifAbsent`?
- It also doesn't explain what `value` is - I'm assuming it's the current value of that key? There could be an example of using the current value to update the map

File: sdk/lib/core/map.dart [hunk 11]
```diff
@@ -261,15 +318,35 @@ abstract class Map<K, V> {
...
   ///
   /// Iterates over all entries in the map and updates them with the result
   /// of invoking [update].
+  /// ```dart
+  /// final data = {1: 'A', 2: 'B', 3: 'C'};
+  /// data.updateAll((key, value) => 'AB');
+  /// print(data); // {1: AB, 2: AB, 3: AB}
+  /// data.updateAll((key, value) => key.toString());
+  /// print(data); // {1: 1, 2: 2, 3: 3}
+  /// ```
```
Comment (tomgilder) [inline sdk/lib/core/map.dart:330]: I suggest adding a couple of line breaks just to split this up
```
   /// ```dart
   /// final data = {1: 'A', 2: 'B', 3: 'C'};
   /// 
   /// data.updateAll((key, value) => 'AB');
   /// print(data); // {1: AB, 2: AB, 3: AB}
   /// 
   /// data.updateAll((key, value) => key.toString());
   /// print(data); // {1: 1, 2: 2, 3: 3}
   /// ```
```

File: sdk/lib/core/map.dart [hunk 12]
```diff
@@ -278,13 +355,14 @@ abstract class Map<K, V> {
...
-  /// for (var key in ['Bob', 'Rohan', 'Sophena']) {
+  /// final scores = {'Bob': 36};
+  /// for (final key in ['Bob', 'Rohan', 'Sophena']) {
   ///   scores.putIfAbsent(key, () => key.length);
   /// }
-  /// scores['Bob'];      // 36
-  /// scores['Rohan'];    //  5
-  /// scores['Sophena'];  //  7
+  /// print(scores); // {Bob: 36, Rohan: 5, Sophena: 7}
+  /// // Return value is current value and no update done if key exists on map
```
Comment (tomgilder) [inline sdk/lib/core/map.dart]:
```suggestion
  /// // If the key already exists, the current value is returned, and the Map isn't modified.
```

Comment (tomgilder) [reply sdk/lib/core/map.dart]: Also wondering if this should be a separate example - or if not, at least separate with a new line

**Review by tomgilder — CHANGES_REQUESTED** [review]

**Review by alexlindroos — APPROVED** [review]

**Review by stepthomp — APPROVED** [review]

**Pull request closed without merging**